### PR TITLE
update SendAsync_ConcurentSendReceive_Ok to use DuplexContent

### LINKED ISSUE #39046: SendAsync_ConcurentSendReceive_Ok test runs very slowly
I often run just the HTTP2 tests with -parallel none and -verbose in order to get a clear picture of what's happening during test execution.

The whole HTTP2 test class takes about 53s to run. Over 30s of this is this configurations of this single test.

It would be good to make this test run more quickly.

### PR DESCRIPTION
**Pull request by wfurt**
I updated this test to use DuplexContent instead of SlowStream. It takes less than .5s to run all 4 variants. It uses synchronization instead of delays to cover duplex and non-duplex processing. 
I also added more comments. 

```
  ~/github/wfurt-corefx-new/artifacts/bin/System.Net.Http.Functional.Tests/netcoreapp-Unix-Debug ~/github/wfurt-corefx-new/src/System.Net.Http/tests/FunctionalTests
    Discovering: System.Net.Http.Functional.Tests (method display = ClassAndMethod, method display options = None)
    Discovered:  System.Net.Http.Functional.Tests (found 1 of 1136 test case)
    Starting:    System.Net.Http.Functional.Tests (parallel test collections = on, max threads = 8)
      System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_Http2.SendAsync_ConcurentSendReceive_Ok(shouldWaitForRequestBody: True, responseCode: Forbidden) [STARTING]
      System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_Http2.SendAsync_ConcurentSendReceive_Ok(shouldWaitForRequestBody: True, responseCode: Forbidden) [FINISHED] Time: 0.3171338s
      System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_Http2.SendAsync_ConcurentSendReceive_Ok(shouldWaitForRequestBody: False, responseCode: Forbidden) [STARTING]
      System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_Http2.SendAsync_ConcurentSendReceive_Ok(shouldWaitForRequestBody: False, responseCode: Forbidden) [FINISHED] Time: 0.0098563s
      System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_Http2.SendAsync_ConcurentSendReceive_Ok(shouldWaitForRequestBody: True, responseCode: OK) [STARTING]
      System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_Http2.SendAsync_ConcurentSendReceive_Ok(shouldWaitForRequestBody: True, responseCode: OK) [FINISHED] Time: 0.0082169s
      System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_Http2.SendAsync_ConcurentSendReceive_Ok(shouldWaitForRequestBody: False, responseCode: OK) [STARTING]
      System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_Http2.SendAsync_ConcurentSendReceive_Ok(shouldWaitForRequestBody: False, responseCode: OK) [FINISHED] Time: 0.0064049s
    Finished:    System.Net.Http.Functional.Tests
  === TEST EXECUTION SUMMARY ===
     System.Net.Http.Functional.Tests  Total: 4, Errors: 0, Failed: 0, Skipped: 0, Time: 0.511s
```

fixes #39046

### CODE REVIEW

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs [hunk 1]
```diff
@@ -2707,54 +2707,69 @@ public async Task PostAsyncNonDuplex_ServerCompletesResponseBodyThenResetsStream
...
-        public async Task SendAsync_ConcurentSendReceive_Ok(bool shouldWait, HttpStatusCode responseCode)
+        public async Task SendAsync_ConcurentSendReceive_Ok(bool shouldWaitForRequestBody, HttpStatusCode responseCode)
         {
-            TaskCompletionSource<bool> tsc = new TaskCompletionSource<bool>();
             string requestContent = new string('*', 300);
             const string responseContent = "SendAsync_ConcurentSendReceive_Ok";
-            var stream = new CustomContent.SlowTestStream(Encoding.UTF8.GetBytes(requestContent), tsc, trigger : 1, count : 10);
 
-            await Http2LoopbackServer.CreateClientAndServerAsync(async url =>
+            using (var server = Http2LoopbackServer.CreateServer())
```
Comment (davidsh) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs]: nit: shouldn't use 'var' in this case. it is not obvious what the type is.

**Review by davidsh — APPROVED** [review]

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs (hunk 1 above)
Comment (wfurt) [reply src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs]: I'll do separate pass to cleanup other instances as well.

**Comment by wfurt** [comment]
windows outerloop failures are unrelated.

**Pull request merged**